src/_gettsim/transfers/erziehungsgeld.py

```diff
@@ -0,0 +1,330 @@
+    erziehungsgeld_params: dict,
+    alleinerz_tu: bool,
+    eink_st_abzuege_params: dict,
+    erziehungsgeld_anspruch_kind: bool,
+) -> float:
+    """Calculate the relevant wage for parental leave benefit (erziehungsgeld) on child
+    level
+
+    legal reference: Bundesgesetzblatt Jahrgang 2004 Teil I Nr. 6 (p.209)
+
+    Parameters
+    ----------
+    bruttolohn_vorj_m_tu
+        See :func:`bruttolohn_vorj_m_tu`.
+    erziehungsgeld_params
+        See params documentation :ref:`erziehungsgeld_params <erziehungsgeld_params>`.
+    alleinerz_tu
+        See :func:`alleinerz_tu`.
+    eink_st_abzuege_params
+        See params documentation :ref:`eink_st_abzuege_params <eink_st_abzuege_params>`.
+    erziehungsgeld_anspruch_kind
+        See :func:`erziehungsgeld_anspruch_kind`.
+
+    Returns
+    -------
+    relevant wage for calculation parental leave benefit (erziehungsgeld)
+
+
+    Todo: There is special rule for "Beamte, Soldaten und Richter" which is not
+          implemented yet
+    """
+
+    if erziehungsgeld_anspruch_kind and alleinerz_tu:
+        out = (
+            bruttolohn_vorj_m_tu * 12
+            - eink_st_abzuege_params["werbungskostenpauschale"]
+        ) * erziehungsgeld_params["pauschal_abzug_auf_einkommen"]
+    elif erziehungsgeld_anspruch_kind and (not alleinerz_tu):
+        out = (
+            bruttolohn_vorj_m_tu * 12
+            - eink_st_abzuege_params["werbungskostenpauschale"]
+            * 2  # is that already hard coded?
+        ) * erziehungsgeld_params["pauschal_abzug_auf_einkommen"]
+    else:
+        out = 0.0
+    return out
+
+
+def erziehungsgeld_einkgrenze_kind(  # noqa: PLR0913
+    erziehungsgeld_params: dict,
+    alleinerz_tu: bool,
+    alter_monate: float,
+    anz_kinder_mit_kindergeld_tu: float,
+    budget_erz_geld: bool,
+    erziehungsgeld_anspruch_kind: bool,
+) -> float:
+    """Calculating the income threshold for parental leave benefit (erziehungsgeld)
+    on child level
+
+    legal reference: Bundesgesetzblatt Jahrgang 2004 Teil I Nr. 6 (pp.208)
+
+    Parameters
+    ----------
+    erziehungsgeld_params
+        See params documentation :ref:`erziehungsgeld_params <erziehungsgeld_params>`.
+    alleinerz_tu
+        See :func:`alleinerz_tu`.
+    alter_monate
+        See :func:`alter_monate`.
+    anz_kinder_mit_kindergeld_tu
+        See :func:`anz_kinder_mit_kindergeld_tu`.
+    budget_erz_geld
+        See :See basic input variable :ref:`budget_erz_geld <budget_erz_geld>`.
+    erziehungsgeld_anspruch_kind
+        See :func:`erziehungsgeld_anspruch_kind`.
+
+    Returns
+    -------
+    income threshold for parental leave benefit (erziehungsgeld)
+    """
+    # There are different income threshold depending on the age of the child,
+    # the fact if a person is a single parent, and if regelsatz or budgetsatz is applied
+
+    einkommensgrenze = {
+        (True, True, False): erziehungsgeld_params["einkommensgrenze_bis_6m"][
```

Comment:
Please split that up in nested functions. Where you select in each function based on one bool. Lets zoom about it!